searx/search/checker/background.py

```diff
@@ -0,0 +1,122 @@
+from searx.search.checker import Checker
+from searx.shared import schedule, storage
+
+
+CHECKER_RESULT = 'CHECKER_RESULT'
+running = threading.Lock()
+
+
+def _get_interval(every, error_msg):
+    if isinstance(every, int):
+        every = (every, every)
+    if not isinstance(every, (tuple, list))\
+       or len(every) != 2\
+       or not isinstance(every[0], int)\
+       or not isinstance(every[1], int):
+        raise SearxSettingsException(error_msg, None)
+    return every
+
+
+def _get_every():
+    every = settings.get('checker', {}).get('scheduling', {}).get('every', (300, 1800))
+    return _get_interval(every, 'checker.scheduling.every is not a int or list')
+
+
+def get_result():
+    serialized_result = storage.get_str(CHECKER_RESULT)
+    if serialized_result is not None:
+        return json.loads(serialized_result)
+
+
+def _set_result(result):
+    result['timestamp'] = int(time.time() / 3600) * 3600
+    storage.set_str(CHECKER_RESULT, json.dumps(result))
+
+
+def run():
+    if not running.acquire(blocking=False):
+        return
+    try:
+        logger.info('Starting checker')
+        result = {
+            'status': 'ok',
+            'engines': {}
+        }
+        for name, processor in processors.items():
+            logger.debug('Checking %s engine', name)
+            checker = Checker(processor)
+            checker.run()
+            if checker.test_results.succesfull:
+                result['engines'][name] = {'success': True}
+            else:
+                result['engines'][name] = {'success': False, 'errors': checker.test_results.errors}
+
+        _set_result(result)
+        logger.info('Check done')
+    except Exception:
+        _set_result({'status': 'error'})
+        logger.exception('Error while running the checker')
+    finally:
+        running.release()
+
+
+def _run_with_delay():
+    every = _get_every()
+    delay = random.randint(0, every[1] - every[0])
+    logger.debug('Start checker in %i seconds', delay)
+    time.sleep(delay)
+    run()
+
+
+def _start_scheduling():
+    every = _get_every()
+    schedule(every[0], _run_with_delay)
+    run()
+
+
+def _signal_handler(signum, frame):
+    t = threading.Thread(target=run)
+    t.daemon = True
+    t.start()
+
+
+def initialize():
+    logger.info('Send SIGUSR1 signal to pid %i to start the checker', os.getpid())
+    signal.signal(signal.SIGUSR1, _signal_handler)
+
+    # disabled by default
+    _set_result({'status': 'disabled'})
+
+    # special case when debug is activate
+    if searx_debug and settings.get('checker', {}).get('off_when_debug', True):
+        logger.info('debug mode: checker is disabled')
+        return
+
+    # check value of checker.scheduling.every now
+    scheduling = settings.get('checker', {}).get('scheduling', None)
+    if scheduling is None or not scheduling:
+        logger.info('Checker scheduler is disabled')
+        return
+
+    #
+    _set_result({'status': 'unknow'})
```

Comment:
On my searx private instance _http://MYLOCALIP/searx/stats/checker_, I have:

{"status":"unknow","timestamp":1610398800}

Typo: should be "unknown".

Reply:
And when I try to refresh this _checker_ web page multiple times during the same minute, I can get various results: _error_, **unknow**, and sometimes the results.
I tried with in Chrome Incognito Window, and normal window.

Reply:
It should be : 
* `unknow(n)` when searx starts.
* then `ok` after an success run of the checker (with or without engine errors)
* or `error` after an exception ( [background.py](https://github.com/searx/searx/blob/eca200404d195fcfaf29a8b30f3cc7d2c7e4c021/searx/search/checker/background.py#L69) )

but different values means an issue.

This code has run successfully:
https://github.com/searx/searx/blob/eca200404d195fcfaf29a8b30f3cc7d2c7e4c021/searx/shared/__init__.py#L15-L17

So most probably there is an issue with [searx.shared.shared_uwsgi](https://github.com/searx/searx/blob/eca200404d195fcfaf29a8b30f3cc7d2c7e4c021/searx/shared/shared_uwsgi.py) or at least with the configuration.

Can you have a look to your uwsgi log ?

Reply:
Ok I have finally added in`/etc/uwsgi/vassals/searx.ini`: 

> # Cache
> cache2 = name=searxcache,items=2000,blocks=2000,blocksize=4096,bitmap=1

It works now.

Reply:
And I add this in my log.

> janv. 11 21:11:51 arch-server uwsgi[23161]: ERROR:searx.shared:uwsgi.ini configuration error, add this line to your uwsgi.ini
> janv. 11 21:11:51 arch-server uwsgi[23161]: cache2 = name=searxcache,items=2000,blocks=2000,blocksize=4096,bitmap=1

Which is now fixed.